sormas-backend/src/main/java/de/symeda/sormas/backend/util/MethodDurationLogger.java

```diff
@@ -0,0 +1,69 @@
+package de.symeda.sormas.backend.util;
+
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.InvocationContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.symeda.sormas.api.utils.DateHelper;
+
+/**
+ * <p>
+ * Purpose: Log duration of method invocation, usually to be to detect performance leaks in the EJB call stack.<br />
+ * You can activate logging with DEBUG level to get all durations over {@value #SIGNIFANT_DURATION} ms.
+ * To see all EJB method starts with parameters and all durations, set log level to TRACE.
+ * </p>
+ * <p>
+ * <b>Warning:</b> Activating this logging has some performance impact itself, so only active it temporarly.
+ * </p>
+ * 
+ * @see https://www.adam-bien.com/roller/abien/entry/simplest_possible_ejb_3_12
+ */
+public class MethodDurationLogger {
+
+	/**
+	 * Avoids fast but noisy and mostly internal EJB calls to be logged on DEBUG level.
+	 */
+	private static final int SIGNIFANT_DURATION = 5;
+
+	private final Logger logger = LoggerFactory.getLogger(MethodDurationLogger.class);
+
+	@AroundInvoke
+	public Object logInvokeDuration(InvocationContext context) throws Exception {
+
+		if (!logger.isDebugEnabled()) {
+			// Directly proceed if logging is not activated
+			return context.proceed();
+		}
+
+		long startTime = DateHelper.startTime();
```

Comment:
Maybe the timer should be started after the initial trace logging.

Reply:
Yes, good idea. No need to measure Logger overhead